sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClientIntegrationTest.java

```diff
@@ -953,6 +960,105 @@ void renewMessageLock(MessagingEntityType entityType) throws InterruptedExceptio
+        footer.put("footer-key-1", "footer-value-1");
+        footer.put("footer-key-2", "footer-value-2");
+
+        final Map<String, Object> aplicaitonProperties = new HashMap<>();
+        aplicaitonProperties.put("ap-key-1", "ap-value-1");
+        aplicaitonProperties.put("ap-key-2", "ap-value-2");
+
+        final Map<String, Object> deliveryAnnotation = new HashMap<>();
+        deliveryAnnotation.put("delivery-annotations-key-1", "delivery-annotations-value-1");
+        deliveryAnnotation.put("delivery-annotations-key-2", "delivery-annotations-value-2");
+
+        setSenderAndReceiver(entityType, TestUtils.USE_CASE_VALIDATE_AMQP_PROPERTIES, isSessionEnabled);
+
+        final String messageId = UUID.randomUUID().toString();
+        final AmqpAnnotatedMessage expectedAmqpProperties = new AmqpAnnotatedMessage(new AmqpDataBody(Collections.singletonList(new BinaryData(CONTENTS_BYTES))));
+        expectedAmqpProperties.getProperties().setSubject(subject);
+        expectedAmqpProperties.getProperties().setReplyToGroupId("r-gid");
+        expectedAmqpProperties.getProperties().setReplyTo("replyto");
+        expectedAmqpProperties.getProperties().setContentType("content-type");
+        expectedAmqpProperties.getProperties().setCorrelationId("corelation-id");
+        expectedAmqpProperties.getProperties().setTo("to");
+        expectedAmqpProperties.getProperties().setAbsoluteExpiryTime(OffsetDateTime.now().plusSeconds(60));
+        expectedAmqpProperties.getProperties().setUserId("user-id-1".getBytes());
+        expectedAmqpProperties.getProperties().setContentEncoding("string");
+        expectedAmqpProperties.getProperties().setGroupSequence(Long.valueOf(2));
+        expectedAmqpProperties.getProperties().setCreationTime(OffsetDateTime.now().plusSeconds(30));
+
+        expectedAmqpProperties.getHeader().setPriority(Short.valueOf((short) 2));
+        expectedAmqpProperties.getHeader().setFirstAcquirer(true);
+        expectedAmqpProperties.getHeader().setDurable(true);
+
+        expectedAmqpProperties.getFooter().putAll(footer);
+        expectedAmqpProperties.getDeliveryAnnotations().putAll(deliveryAnnotation);
+        expectedAmqpProperties.getApplicationProperties().putAll(aplicaitonProperties);
+
+        final ServiceBusMessage message = getMessage(messageId, isSessionEnabled, expectedAmqpProperties);
+
+        sendMessage(message).block(TIMEOUT);
+
+        StepVerifier.create(receiver.receiveMessages().map(ServiceBusReceivedMessageContext::getMessage))
+            .assertNext(received -> {
+                assertNotNull(received.getLockToken());
+                AmqpAnnotatedMessage actual = received.getAmqpAnnotatedMessage();
+                try {
+                    assertArrayEquals(CONTENTS_BYTES, message.getBody());
+                    assertEquals(expectedAmqpProperties.getHeader().getPriority(), actual.getHeader().getPriority(), "Header.priority is not equal.");
```

Comment:
The extra message is not necessary. If this fails, we'll know the line that it failed at.